blueprints/controllers/ikea_e1744/ikea_e1744.yaml

```
@@ -194,18 +217,21 @@ action:
         {%- elif integration_id == "zha" -%}
         {{ trigger.event.data.command }}{{"_" if trigger.event.data.args|length > 0}}{{ trigger.event.data.args|join("_") }}
         {%- endif -%}
+      last_controller_event: '{%- if "trigger_action" in states(helper_last_controller_event) -%}
+          {{ (states(helper_last_controller_event) | from_json).trigger_action }}
+        {%- endif -%}'
+  # save the last controller event, store the button event
+  - choose:
+    - conditions: '{{trigger_action != last_controller_event}}'
+      sequence:
+      - service: input_text.set_value
+        data:
+          entity_id: !input 'helper_last_controller_event'
+          value: '{"trigger_action":"{{trigger_action}}","last_triggered":"{{now()}}","last_event":"{{last_controller_event}}" }' 
```

```suggestion
  - service: input_text.set_value
    data:
      entity_id: !input 'helper_last_controller_event'
      value: '{"trigger_action":"{{trigger_action}}","last_triggered":"{{now()}}","last_event":"{{last_controller_event}}" }' 
```

Don't have a proper way to test this, but I guess that if we store the object only when the trigger is different we don't always get debouncing (if a user only clicks the remote, the `last_triggered` value will always be the one from the first press).

---

No, i need the last_triggered value to always be updated for debouncing and such.

---

Sorry, previous comment was nonsense. The reason I do it like this is because else the wait for trigger also causes a restart.

---

Ok don't worry. We'll look into this later. :)

---

I remember now! since we ignore double events we also don't want to register them in the helper, because that influences the last_triggered time when it's actually ignored.